test/clj_sqs_extended/core_test.clj

```diff
@@ -111,6 +114,61 @@
           (is (contains? stats :stopped-at))))
       (close! handler-chan))))
 
+(deftest handle-queue-terminates-after-restart-count-exceeded
+  (testing "handle-queue terminates when the restart-count exceeds the limit"
+    (let [handler-chan (chan)]
+      (fixtures/with-test-standard-queue
+        (with-redefs-fn {#'sqs/receive-message
+                         (fn [_ _ _] (HttpTimeoutException.
+                                       "Something seems to be wrong with the network."))}
+          #(let [stop-fn (fixtures/with-handle-queue-queue-opts-standard-no-autostop
+                           handler-chan
+                           {:restart-limit         3
+                            :restart-delay-seconds 1})]
+             (Thread/sleep 3500)
```

Comment:
I'm not a fan of these hardcoded waits. I see why you need them because this is async. All tests for a simple library like this should finish within a couple seconds _ideally_ 😅. Consider using our `wait-for` library. Could you create a new issue for v0.2?

Also, perhaps make `restart-limit` a var in the let above so that you can refer to it here too? i.e. `(Thread/sleep (+ (* restart-limit 1000) 500)`

Reply:
Yes, good point. I'll create an issue to improve this later on with v0.2 then.